pkg/apis/elasticsearch/v1/elasticsearch_types.go

```diff
@@ -459,17 +464,70 @@ func (es *Elasticsearch) ElasticServiceAccount() (commonv1.ServiceAccountName, e
 	return "", nil
 }
 
-// IsAutoscalingDefined returns true if there is an autoscaling configuration in the annotations.
-func (es Elasticsearch) IsAutoscalingDefined() bool {
+// IsAutoscalingAnnotationSet returns true if there is an autoscaling configuration in the annotations.
+// Deprecated: the autoscaling annotation has been deprecated in favor of the ElasticsearchAutoscaler custom resource.
+func (es Elasticsearch) IsAutoscalingAnnotationSet() bool {
 	_, ok := es.Annotations[ElasticsearchAutoscalingSpecAnnotationName]
 	return ok
 }
 
-// AutoscalingSpec returns the autoscaling spec in the Elasticsearch manifest.
-func (es Elasticsearch) AutoscalingSpec() string {
+// AutoscalingAnnotation returns the autoscaling spec in the Elasticsearch manifest.
+// Deprecated: the autoscaling annotation has been deprecated in favor of the ElasticsearchAutoscaler custom resource.
+func (es Elasticsearch) AutoscalingAnnotation() string {
 	return es.Annotations[ElasticsearchAutoscalingSpecAnnotationName]
 }
 
+var _ v1alpha1.AutoscalingResource = &Elasticsearch{}
+
+// AutoscalingSpec is the root object of the autoscaling specification in the Elasticsearch resource definition.
+// +kubebuilder:object:generate=false
+type AutoscalingSpec struct {
+	AutoscalingPolicySpecs v1alpha1.AutoscalingPolicySpecs `json:"policies"`
+
+	// PollingPeriod is the period at which to synchronize and poll the Elasticsearch autoscaling API.
+	PollingPeriod *metav1.Duration `json:"pollingPeriod"`
+
+	// Elasticsearch is stored in the autoscaling spec for convenience. It should be removed once the autoscaling spec is
+	// fully part of the Elasticsearch specification.
+	Elasticsearch Elasticsearch `json:"-"`
+}
+
+func (es Elasticsearch) GetAutoscalingPolicySpecs() (v1alpha1.AutoscalingPolicySpecs, error) {
+	autoscalingSpecification, err := es.GetAutoscalingSpecificationFromAnnotation()
+	if err != nil {
+		return nil, err
+	}
+	return autoscalingSpecification.AutoscalingPolicySpecs, nil
+}
+
+func (es Elasticsearch) GetPollingPeriod() (*metav1.Duration, error) {
+	autoscalingSpecification, err := es.GetAutoscalingSpecificationFromAnnotation()
+	if err != nil {
+		return nil, err
+	}
+	return autoscalingSpecification.PollingPeriod, nil
+}
+
+func (es Elasticsearch) GetElasticsearchAutoscalerStatus() (v1alpha1.ElasticsearchAutoscalerStatus, error) {
+	autoscalingStatus, err := ElasticsearchAutoscalerStatusFrom(es)
+	if err != nil {
+		return v1alpha1.ElasticsearchAutoscalerStatus{}, err
+	}
+	return autoscalingStatus, nil
```

Comment:
```suggestion
	return ElasticsearchAutoscalerStatusFrom(es)
```

Reply:
It's done on purpose to be sure we do not return a half built/ corrupted status.